src/System.Private.Xml/tests/Writers/XmlWriterApi/TCFullEndElement.cs

```diff
@@ -4270,7 +4269,8 @@ public void writeValue_27(XmlWriterUtils utils, int param, string sourceStr, str
                     if (param == 1)
                         w.WriteValue(value[sourceStr]);
                     else
-                        w.WriteAttributeString("a", value[sourceStr].ToString());
+                        w.WriteAttributeString("a",
+                            string.Format(CultureInfo.InvariantCulture, "{0}", value[sourceStr]));
```

Comment:
In the try block below, it is setting CultureInfo.CurrentCulture to InvariantCulture.
Wouldn't it be cleaner if you move these two lines from `try` block:

```c#
origCulture = CultureInfo.CurrentCulture;
CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;  // So that the number format doesn't depend on the current culture
```

before the `using` block without touching this line?

Reply:
Such optimization is bug because `w.WriteValue(value[sourceStr]);` should call inside current culture to test it correctly.